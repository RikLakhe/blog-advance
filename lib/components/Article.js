import React from 'react';

const styles= {
    article:{
        paddingBottom: 10,
        borderBottomStyle: 'solid',
        borderBottomColor: '#aaa',
        borderBottomWidth: 1,
        marginBottom: 10,
    }
};

const dateDisplay = dateString => new Date(dateString).toDateString();

const Article = (props) => {
    const {
        article,
        actions,
    } = props;

    const author = actions.lookupAuthor(article.authorId);
    
    return (
        <div>
            <div style={styles.article}>{article.title}</div>
            <div>{dateDisplay(article.date)}</div>
            <div><a href={author.website}>{author.firstName} {author.lastName}</a></div>
            <div>{article.body}</div>
        </div>
    );
};

export default Article;
