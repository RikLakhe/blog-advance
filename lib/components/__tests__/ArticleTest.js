import React from 'react';
import Article from '../Article';

import renderer from 'react-test-renderer';

describe('Article', () => {
    let tree;
    beforeAll(() => {
        const testProps = {
            article : {id:'a',authorId:'testauthorid',title:'titletest',date:'date',body:'bodytest'},
            actions : {
                lookupAuthor : jest.fn(()=>({ firstName: 'test', lastName:'testlast',website:'http://test.com'}))
            }
        };

        tree = renderer.create(<Article {...testProps} />).toJSON();
    });


    it('renders correctly',()=>{
        expect(tree.children.length).toBe(4);
        expect(tree).toMatchSnapshot();
    });
});
