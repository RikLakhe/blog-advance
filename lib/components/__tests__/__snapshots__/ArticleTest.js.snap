// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders correctly 1`] = `
<div>
  <div
    style={
      Object {
        "borderBottomColor": "#aaa",
        "borderBottomStyle": "solid",
        "borderBottomWidth": 1,
        "marginBottom": 10,
        "paddingBottom": 10,
      }
    }
  >
    titletest
  </div>
  <div>
    Invalid Date
  </div>
  <div>
    <a
      href="http://test.com"
    >
      test
       
      testlast
    </a>
  </div>
  <div>
    bodytest
  </div>
</div>
`;
